src/graphics/reproject-texture.js

```diff
@@ -332,72 +339,46 @@ class SimpleCache {
         }
         return this.map.get(key);
     }
-
-    clear() {
-        this.map.clear();
-    }
 }
 
-// per-device cache
-class DeviceCache {
-    constructor() {
-        this.cache = new SimpleCache();
-    }
+// cache, used to store:
+// - samples. we store these separately from textures since multiple devices can use the same
+//   set of samples.
+// - texures inside DeviceResourceCache.reprojectTextureCache - those are per device
+const samplesCache = new SimpleCache();
 
-    // get the cache entry for the given device and key
-    // if entry doesn't exist, missFunc will be invoked to create it
-    get(device, key, missFunc) {
-        return this.cache.get(device, () => {
-            const cache = new SimpleCache();
-            device.on('destroy', () => {
-                cache.map.forEach((value, key) => {
-                    value.destroy();
-                });
-                this.cache.map.delete(device);
-            });
-            return cache;
-        }).get(key, missFunc);
+const getCachedTexture = (device, key, getSamplesFnc) => {
+    const cache = DeviceResourceCache.get(device);
+    if (!cache.reprojectTextureCache) {
+        cache.reprojectTextureCache = new SimpleCache();
     }
-
-    clear() {
-        this.cache.clear();
+    if (!cache.reprojectTextureCache.map.get(key)) {
+        const texture = createSamplesTex(device, key, samplesCache.get(key, () => {
+            return getSamplesFnc();
+        }));
```

Comment:
Looks like an empty lambda to me that is just adding overhead/code:

```suggestion
        const texture = createSamplesTex(device, key, samplesCache.get(key, getSamplesFnc));
```

Reply:
done too, thanks!